ntoskrnl/ke/i386/mp.c

```diff
@@ -118,7 +126,10 @@ KeStartAllProcessors()
         /* Finally call HalStartNextProcessor with the prepped values */
         ApplicationProcessorsEnabled = HalStartNextProcessor(KeLoaderBlock, &ProcessorState);
 
-        if(ApplicationProcessorsEnabled == FALSE){
+        if (ApplicationProcessorsEnabled == FALSE){
+            ProcessorCount--;
+            /* Started means the AP itself is online, this doesn't mean it's seen by the kernel */
+            DPRINT1("HalStartNextProcessor: Sucessful AP startup count is %X\n", ProcessorCount);
```

Comment:
I'm confused about what this code is about. Is it failure or success ?

Reply:
It's an intended failure. We continue the loop until processors are no longer being successfully started

Reply:
Then I think I don't understand the comment above.

Reply:
Which comment? also this If condition isn't in the file anymore